### DISCUSSION TITLE
How can I stop reacting to a signal update?

### DISCUSSION THREAD

**Discussion by danieltroger** [solidjs/solid#655] [Q&A] [answered]
I basically just want to know whether a signal has turned `true` once, then do a thing (resolve a promise which triggers more stuff) but then never care about the change of that signal from that place in the code again.

I'm using `createComputed` since I think that was what reacts most instantly, so how can I stop the inner function in `createComputed` from executing again?

**Comment by atk** [comment, marked as answer]
You can use a derived memo:

```js
const [source, setSource] = createSignal(false);
let switchDone = false;
const target = createMemo(() => switchDone || (switchDone = source()));
```

Update: since source() will only be called off switchDone is false, we can omit untrack.

**Reply by danieltroger** [reply]
@atk sorry, I don't understand how your example code would work.

Edit: without untrack I guess it makes sense, wouldn't untrack only have made it work if the signal was already set?

I think I asked the question badly, because I'm not helped with a variable switching or another signal only being set once.

This is what I have right now:

```
const [show, set_show] = createSignal(false); // whether dialog should be shown


// wait for dialog to have been opened
await new Promise(r => {
  createComputed(() => {
    if (show()) {
      r();
    }
  });
});
```


Actually I think I understand your idea, setting a variable so that when the thing is reactively called the second time it's no longer reactive and gets untracked or something?

**Reply by danieltroger** [reply]
@atk would this still stop tracking after the first time?

```
await new Promise(r => {
  let has_resolved = false;
  createComputed(() => {
    if (!has_resolved && show()) {
      r();
      has_resolved = true;
    }
  });
});
```

**Comment by ryansolid** [comment]
~~You don't want to create a computed in a promise. Write to a signal. All reactive primitives that track should be created synchronously as part of reactive execution (ie.. under a root or parent reactive context).~~

Nvm, the promise constructor runs synchronously. Don't mind me.

**Reply by atk** [reply]
At some point you need a bridge between the reactive context and your target that only reacts once. This could be a memo or an effect setting a signal (which is basically the same, the latter just the long way to describe the former).
